src/core/version_neg.h

```diff
@@ -0,0 +1,100 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+Abstract:
+
+    This file contains definitions for version negotiation.
+
+--*/
+
+#pragma once
+
+//
+// This list is the versions that the server advertises support for.
+//
+extern const uint32_t DefaultSupportedVersionsList[];
```

Comment:
It's a bit more painful to update, but I prefer to put the size here as a constant, and then we don't need the length field.